HW6/HW6.cpp

```diff
@@ -0,0 +1,196 @@
+
+    std::cout << "Enter positive integer: ";
+    std::cin >> n;
+    while (n < 0) {
+        std::cout << "Entered integer isn't positive!\n" << "Enter positive integer again : ";
+        std::cin >> n;
+    }
+
+    for (int i = 1; i <= n; i++) {
+        factorial *= i;  
+    }
+    std::cout << "Factorial of " << n << " is " << factorial;
+   }
+void geometricFigures() {
+    int n = 0;
+    std::cout << "Enter positive value: ";
+    std::cin >> n;
+    while (n < 1) {
+        std::cout << "Entered value isn't positive!\n" << "Enter positive value again : ";
+        std::cin >> n;
+    }
+    for (int i = n; i > 0; i--) {
+        for (int j = 0; j < i; j++) {
+            std::cout << "* ";
+        }
+        std::cout << std::endl;
+    }
+    int m = 0;
+    std::cout << "Enter positive value: ";
+    std::cin >> m;
+    while (m < 1) {
+        std::cout << "Entered value isn't positive!\n" << "Enter positive value again : ";
+        std::cin >> m;
+    }
+    for (int i = 0; i < m; i++) {
+        for (int j = 0; j <= i; j++) {
+            std::cout << "* ";
+        }
+        std::cout << std::endl;
+    }
+    int width = 0;
+    std::cout << "Enter width: ";
+    std::cin >> width;
+    int height = 0;
+    std::cout << "Enter height: ";
+    std::cin >> height;
+    if (width <= 0 && height <=0) {
+        std::cout << "Sorry, but we can't build figure without positive value!";
+    }
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            std::cout << "* ";
+        }
+        std::cout << std::endl;
+    }
+    int l = 0;
+    std::cout << "Enter positive value: ";
+    std::cin >> l;
+    while (l < 1) {
+        std::cout << "Entered value isn't positive!\n" << "Enter positive value again : ";
+        std::cin >> l;
+    }
+    for (int i = 0; i <l; i++) {
+       
+        for (int j = 0; j < i; j++) {
+            std::cout << " ";
+        }
+        for (int k = 0; k <l; k++) {
+            std::cout << "* ";
+        }
+        std::cout << std::endl;
+    }
+    int t = 0;
+    std::cout << "Enter positive value: ";
+    std::cin >> t;
+    while (t < 1) {
+        std::cout << "Entered value isn't positive!\n" << "Enter positive value again : ";
+        std::cin >> t;
+    }
+    for (int i = 0; i < t; i++) {
+        for (int j = 0; j <= i; j++) {
+            if (i % 2 == 0) {
+                if (j % 2 == 0) { std::cout << 1 << " "; }
+                else { std::cout << 0 << " "; }
+            }
+            else{
+                if (j % 2 == 0) { std::cout << 0 << " "; }
+                else { std::cout << 1 << " "; }
+            }
+        }
+        std::cout << std::endl;
+
+    }
+}
+void charManip() {
+    char value = 0;
+    int totalSum = 0;
+
+    while (true) {
+        std::cout << "Enter value: ";
+        std::cin >> value;
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+        if (value > 96 && value < 123) {
```

Comment:
96 and 123 is out of context here. Can compare symbols directly, e.g. value > 'a'